File: 프로그래머스/lv2/12941. 최솟값 만들기/최솟값 만들기.cpp
#include <iostream>
#include <vector>
#include <algorithm>

using namespace std;

int solution(vector<int> A, vector<int> B) {
    int answer = 0;

    sort(A.begin(), A.end());
    sort(B.begin(), B.end(), greater<int>());
    
    int length = A.size();
    
    for (int i = 0; i < length; i++) {
        int a = A[i];
        int b = B[i];
        
        answer += a * b;
    }
    
    return answer;
}

File: 프로그래머스/2/12941. 최솟값 만들기/최솟값 만들기.java
import java.util.*;
import java.io.*;

class Solution {
    public int solution(int[] A, int[] B) {
        int answer = 0;
        int size = A.length;
        
        Integer[] arrA = new Integer[A.length];
        Integer[] arrB = new Integer[B.length];
        
        for (int i = 0; i < size; i++) {
            arrA[i] = A[i];
            arrB[i] = B[i];
        }
        
        Arrays.sort(arrA);
        Arrays.sort(arrB, Collections.reverseOrder());

        int result1 = 0;

        for (int i = 0; i < size; i++) {
            int a = arrA[i];
            int b = arrB[i];

            result1 += a * b;
        }

        Arrays.sort(arrA, Collections.reverseOrder());
        Arrays.sort(arrB);

        int result2 = 0;

        for (int i = 0; i < size; i++) {
            int a = arrA[i];
            int b = arrB[i];

            result2 += a * b;
        }

        answer = Math.min(result1, result2);

        return answer;
    }
}